reference/services/buildbot.rst

```diff
@@ -0,0 +1,158 @@
+You can access our hosted instance at http://lpbuildbot.canonical.com
+
+Detailed description
+--------------------
+
+There are the following main components in the Launchpad Buildbot
+setup. 
+
+1. `lpbuildbot <https://code.launchpad.net/~canonical-launchpad-branches/lpbuildbot/>`__:
+   This contains the \`master.cfg\` file and the related Python code
+   which powers the
+   `buildmaster <https://docs.buildbot.net/0.8.12/manual/introduction.html>`__
+   in our buildbot setup. There is a public bazaar branch and a private
+   production branch, which contains some secrets related to our
+   production deployment. We add changes to the public branch via merge
+   proposals and then merge them back to the private “production” branch
+   before deployment. In the ``master.cfg`` file, we have defined all
+   the necessary configurations (schedulers, workers, job steps) for
+   buildbot. 
+
+   1. The buildmaster has a custom ``buildbot-poll.py`` cron script
+      which is responsible for merging the commits in the
+      \`master`/\`db-devel\` branches, that have completed a successful
+      buildbot run, to the \`stable`/\`db-stable\` branches
+      respectively. It also periodically merges the changes from the
+      \`stable\` branch to the \`db-devel\` branch, which then
+      propagates to the \`db-stable\` branch after a successful buildbot
+      run. For this, the secrets to push to these branches are present in the buildmaster
+      unit.
+
+ 
+
+2. `Buildbot Worker Charm and Deb
+   Package <https://launchpad.net/~launchpad/lpbuildbot-worker/+charm/lpbuildbot-worker>`__:
+   This repository contains the code for the
+   `buildslave <https://docs.buildbot.net/0.8.12/manual/introduction.html>`__
+   which is deployed using the `lpbuildbot-worker
+   charm <https://charmhub.io/lpbuildbot-worker/>`__ and the
+   `lpbuildbot-worker debian
+   package <https://code.launchpad.net/~launchpad/+archive/ubuntu/ppa/+packages?field.name_filter=buildbot-worker&field.status_filter=published&field.series_filter=>`__.
+   Recipes to build these are present on the `lpbuildbot-worker
+   project <https://launchpad.net/lpbuildbot-worker>`__ in Launchpad.
+
+   1. Working directories for jobs runs for vanilla images and flavors
+      can be found at ``/var/lib/buildbot/slaves`` 
+
+Documentation
+-------------
+
+- `InformationInfrastructure/OSA/LPHowTo/Buildbot - Canonical
+  Wiki <https://wiki.canonical.com/InformationInfrastructure/OSA/LPHowTo/Buildbot>`__
+
+- https://wiki.canonical.com/Launchpad/Buildbot#code_references
+
+Git repositories
+----------------
+
+- https://code.launchpad.net/lpbuildbot-worker
+
+- https://launchpad.net/~launchpad/lpbuildbot-worker/+charm/lpbuildbot-worker
+
+Deployment
+----------
+
+To get shell access to Launchpad Buildbot deployment. You can follow the
+following steps. 
+
+1. SSH into your bastion account.
+
+2. Run ``pe`` and select the corresponding number ID of the buildbot
```

Comment:
Maybe we could just tell to run `sudo -iu stg-launchpad-buildbot`, it might be more in line with other docs and it makes it clearer which account to use